indico/web/client/js/react/components/PublicationSwitch.jsx

```diff
@@ -0,0 +1,137 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import publicationURL from 'indico-url:contributions.manage_publication';
+
+import React, {useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {Header, Modal, Button, Checkbox, List} from 'semantic-ui-react';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {Translate} from 'indico/react/i18n';
+
+export default function PublicationSwitch({eventId}) {
+  const [published, setPublished] = useState(true);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchPublicationSetting = async () => {
+      let response;
+      try {
+        response = await indicoAxios.get(publicationURL({confId: eventId}));
+      } catch (error) {
+        handleAxiosError(error);
+      }
+      setPublished(response.data.published);
+    };
+    fetchPublicationSetting();
+  }, [eventId]);
+
+  const changeState = shouldPublish => {
+    const updatePublicationSetting = async () => {
```

Comment:
No need for this nesting. Just make `changeState` async. We only need this kind of wrapping in useEffect, because the effect's function should not be returning a promise (which all async functions do)